extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/v1/GreenbidsRealTimeDataProcessedAuctionRequestHook.java

```diff
@@ -0,0 +1,219 @@
+
+    private static final String CODE = "greenbids-real-time-data-processed-auction-request-hook";
+    private static final String ACTIVITY = "greenbids-filter";
+    private static final String SUCCESS_STATUS = "success";
+    private static final String BID_REQUEST_ANALYTICS_EXTENSION_NAME = "greenbids-rtd";
+
+    private final ObjectMapper mapper;
+    private final JacksonMapper jacksonMapper;
+    private final Cache<String, OnnxModelRunner> modelCacheWithExpiration;
+    private final Cache<String, ThrottlingThresholds> thresholdsCacheWithExpiration;
+    private final String gcsBucketName;
+    private final String onnxModelCacheKeyPrefix;
+    private final String thresholdsCacheKeyPrefix;
+    private final Storage storage;
+    private final File database;
+
+    public GreenbidsRealTimeDataProcessedAuctionRequestHook(
+            ObjectMapper mapper,
+            Cache<String, OnnxModelRunner> modelCacheWithExpiration,
+            Cache<String, ThrottlingThresholds> thresholdsCacheWithExpiration,
+            String gcsBucketName,
+            String onnxModelCacheKeyPrefix,
+            String thresholdsCacheKeyPrefix,
+            Storage storage,
+            File database) {
+        this.mapper = Objects.requireNonNull(mapper);
+        this.jacksonMapper = new JacksonMapper(mapper);
+        this.modelCacheWithExpiration = modelCacheWithExpiration;
+        this.thresholdsCacheWithExpiration = thresholdsCacheWithExpiration;
+        this.gcsBucketName = gcsBucketName;
+        this.onnxModelCacheKeyPrefix = onnxModelCacheKeyPrefix;
+        this.thresholdsCacheKeyPrefix = thresholdsCacheKeyPrefix;
+        this.storage = storage;
+        this.database = database;
+    }
+
+    @Override
+    public Future<InvocationResult<AuctionRequestPayload>> call(
+            AuctionRequestPayload auctionRequestPayload,
+            AuctionInvocationContext invocationContext) {
+        final AuctionContext auctionContext = invocationContext.auctionContext();
+        final BidRequest bidRequest = auctionContext.getBidRequest();
+        final Partner partner = parseBidRequestExt(bidRequest);
+
+        if (partner == null) {
+            return Future.succeededFuture(toInvocationResult(
+                    bidRequest, null, InvocationAction.no_action));
+        }
+
+        OnnxModelRunner onnxModelRunner = null;
+        ThrottlingThresholds throttlingThresholds = null;
+        Map<String, Map<String, Boolean>> impsBiddersFilterMap = null;
+        try {
+            final OnnxModelRunnerWithThresholds onnxModelRunnerWithThresholds = new OnnxModelRunnerWithThresholds(
+                    jacksonMapper,
+                    modelCacheWithExpiration,
+                    thresholdsCacheWithExpiration,
+                    storage,
+                    gcsBucketName,
+                    onnxModelCacheKeyPrefix,
+                    thresholdsCacheKeyPrefix);
```

Comment:
do we really need to create it on each call? looks like it uses only the hook fields

Reply:
Have moved onnxModelRunnerWithThresholds to GreenbidsRealTimeDataConfig as bean